### PR TITLE
Add stale flag to indicate tracking data deletion

### PR DESCRIPTION
**Pull request by a5anka**

### CODE REVIEW

**Comment by Asitha** [comment]
When is a message is marked as stale? What happens to a stale message? Just curious :-)

**Comment by a5anka** [comment]
Stale flag is used to mark a message already in the outbound disruptor when all the subscribers for message destination leaves (i.e. when the slot becomes orphaned). This is to avoid messages being duplicated.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/DeliverableAndesMetadata.java [hunk 1]
```diff
@@ -49,6 +49,11 @@
      */
     private long timeMessageIsRead;
 
+    /**
+     * Indicate if the metadata should not be used.
+     */
+    private boolean stale;
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/DeliverableAndesMetadata.java]: shall we explain the usage properly?

Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/DeliverableAndesMetadata.java]: Done in https://github.com/wso2/andes/pull/335/.

**Pull request merged**